publish.js

```diff
@@ -199,57 +223,37 @@ publish.getDevVersion = function (moduleVersion, options) {
  * If isTest is specified, it will instead create a tarball in the local directory.
  *
  * @param isTest {Boolean} - indicates if this is a test run or not
- * @param options {Object} - e.g. {"packCmd": "npm pack", "publishCmd": "npm publish"}
+ * @param isDev {Boolean} - indicates if this is a development (true) or standard (false) release
+ * @param options {Object} - e.g. {"packCmd": "npm pack", "publishCmd": "npm publish", "publishDevCmd": "npm publish --tag", "publishHint": "publish hint", "publishDevHint": "publish dev hint", devTag: "dev"}
  */
-publish.pubImpl = function (isTest, options) {
+publish.pubImpl = function (isTest, isDev, options) {
     if (isTest) {
         // create a local tarball
-        var packCmd = options.packCmd || defaults.packCmd;
-        publish.execSync(packCmd);
+        publish.execSyncFromTemplate(options.packCmd);
     } else {
         // publish to npm
-        var publishCmd = options.publishCmd || defaults.publishCmd;
-        publish.execSync(publishCmd);
+        var pubCmd = isDev ? options.publishDevCmd : options.publishCmd;
+        publish.execSyncFromTemplate(pubCmd, options, options.publishHint);
```

Comment:
It looks like this command doesn't use options.publishDevHint when publishing a dev release, as I imagine it should